exovetter/utils.py

```diff
@@ -0,0 +1,426 @@
+        Cadences where this array is `True` are ignored in the calculation.
+        This is useful if some of the entries of time are NaN.
+
+    Returns
+    -------
+    idx : array_like
+        Array of booleans with the same length as ``time``.
+        Element set to `True` are affected by transits.
+
+    Raises
+    ------
+    ValueError
+        Invalid input.
+
+    """
+    if flags is None:
+        flags = np.zeros_like(time, dtype=np.bool_)
+
+    good_time = time[~flags]
+    i0 = np.floor((np.min(good_time) - epoch_bkjd) / period_days)
+    i1 = np.ceil((np.max(good_time) - epoch_bkjd) / period_days)
+    if not np.isfinite(i0) or not np.isfinite(i1):
+        raise ValueError('Error bounding transit time')
+
+    transit_times = epoch_bkjd + period_days * np.arange(i0, i1 + 1)
+    big_num = sys.float_info.max  # A large value that isn't NaN
+    max_diff = 0.5 * duration_days * num_durations
+
+    idx = np.zeros_like(time, dtype=np.bool8)
+    for tt in transit_times:
+        diff = time - tt
+        diff[flags] = big_num
+        if not np.all(np.isfinite(diff)):
+            raise ValueError('NaN found in diff of transit time')
+
+        idx = np.bitwise_or(idx, np.fabs(diff) < max_diff)
+
+    if not np.any(idx):
+        warnings.warn('No cadences found matching transit locations')
+
+    return idx
+
+
+def median_detrend(flux, nPoints):
+    """Quick and dirty median smooth function.
+    Not designed to be at all effecient"""
+
+    size = len(flux)
+    filtered = np.zeros_like(flux)
+    for i in range(size):
+        lwr = max(i - nPoints, 0)
+        upr = min(lwr + 2 * nPoints, size)
+        lwr = upr - 2 * nPoints
+
+        sub = flux[lwr:upr]
+
+        offset = np.median(sub)
+        filtered[i] = flux[i] - offset
+
+    return filtered
+
+
+def sigmaClip(y, nSigma, maxIter=1e4, initialClip=None):
+    """Iteratively find and remove outliers
+
+    Find outliers by identifiny all points more than **nSigma** from
+    the mean value. The recalculate the mean and std and repeat until
+    no more outliers found.
+
+    Inputs:
+    ----------
+    y
+        (1d numpy array) Array to be cleaned
+    nSigma
+        (float) Threshold to cut at. 5 is typically a good value for
+        most arrays found in practice.
+
+    Optional Inputs:
+    -------------------
+    maxIter
+        (int) Maximum number of iterations
+
+    initialClip
+        (1d boolean array) If an element of initialClip is set to True,
+        that value is treated as a bad value in the first iteration, and
+        not included in the computation of the mean and std.
+
+    Returns:
+    ------------
+    1d numpy array. Where set to True, the corresponding element of y
+    is an outlier.
+    """
+
+    idx = initialClip
+    if initialClip is None:
+        idx = np.zeros(len(y), dtype=bool)
+
+    if (len(idx) != len(y)):
+        raise AssertionError('length of array y not equal to initialClip')
```

Comment:
ValueError is less vague, FWIW.
```suggestion
        raise ValueError('length of array y not equal to initialClip')
```

Reply:
But it isn't a problem with a value, it's a problem with the length of an array (which I know is a value), but as a user I think ValueError would be confusing to raise for this issue.  Is there a length of array error?

Reply:
If you want it to be very specific, you could always subclass an exception and make your own. Up to you.

```python
class LengthOfArrayError(ValueError):
    pass
```

Reply:
That feels like overkill unless we were doing it a lot. Let's just keep it as AssertionError...vague, but not misleading.